backend/src/controllers/volunteerController.ts

```diff
@@ -1 +1,61 @@
-// volunteerController.js
\ No newline at end of file
+import { Request, Response } from 'express';
+import VolunteerModel from '../models/volunteerModel.js';
+
+const volunteerModel = new VolunteerModel();
+
+async function getVolunteerById(req: Request, res: Response) {
+    const { volunteer_id } = req.params;
+    
+    if (!volunteer_id) {
+        return res.status(400).json({
+            error: "Missing required parameter: 'volunteer_id'"
+        });
+    }
+
+    try {
+        const volunteer = await volunteerModel.getVolunteerById(volunteer_id);
+        res.status(200).json(volunteer);
+    } catch (error) {
+        return res.status(500).json({
+            error: "Internal server error"
```

Comment:
When testing the error case here, the custom error messages thrown by VolunteerModel.js were not re thrown here (by the controller functions),  they were overwritten by the "Internal Server Error" messages. Could change the error handling to:

`        return res.status(500).json({
            error: `Internal server error: ${error}`
        });`

to show the error returned from the inner function calls.

Reply:
sure np